enterprise/backup/src/main/java/org/neo4j/backup/impl/ForceCheckpointBackupService.java

```diff
@@ -0,0 +1,132 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.backup.impl;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import org.neo4j.com.storecopy.ExternallyManagedPageCache;
+import org.neo4j.graphdb.factory.GraphDatabaseFactory;
+import org.neo4j.graphdb.factory.GraphDatabaseSettings;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.configuration.Settings;
+import org.neo4j.kernel.impl.enterprise.configuration.OnlineBackupSettings;
+import org.neo4j.kernel.impl.transaction.log.checkpoint.CheckPointer;
+import org.neo4j.kernel.impl.transaction.log.checkpoint.SimpleTriggerInfo;
+import org.neo4j.kernel.impl.transaction.log.files.LogFiles;
+import org.neo4j.kernel.impl.transaction.log.files.LogFilesBuilder;
+import org.neo4j.kernel.impl.transaction.log.pruning.LogPruning;
+import org.neo4j.kernel.impl.transaction.log.rotation.LogRotation;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.kernel.lifecycle.Lifecycle;
+
+class ForceCheckpointBackupService implements Lifecycle
+{
+    private final PageCache pageCache;
+    private final FileSystemAbstraction fileSystemAbstraction;
+    private final Config config;
+    private final File backupDirectory;
+
+    private GraphDatabaseAPI graphDatabaseAPI;
+
+    ForceCheckpointBackupService( PageCache pageCache, FileSystemAbstraction fileSystemAbstraction, Config config, File backupDirectory )
+    {
+        this.pageCache = pageCache;
+        this.fileSystemAbstraction = fileSystemAbstraction;
+        this.config = config;
+        this.backupDirectory = backupDirectory;
+    }
+
+    @Override
+    public void init() throws Throwable
+    {
+    }
+
+    @Override
+    public void start() throws Throwable
+    {
+        Map<String,String> configParams = config.getRaw();
+        configParams.put( GraphDatabaseSettings.logical_logs_location.name(), backupDirectory.toString() );
+        configParams.put( GraphDatabaseSettings.pagecache_warmup_enabled.name(), Settings.FALSE );
+        configParams.put( OnlineBackupSettings.online_backup_enabled.name(), Settings.FALSE );
+        configParams.put( GraphDatabaseSettings.keep_logical_logs.name(), Settings.FALSE );
+        GraphDatabaseFactory factory = ExternallyManagedPageCache.graphDatabaseFactoryWithPageCache( pageCache );
+        graphDatabaseAPI = (GraphDatabaseAPI) factory.newEmbeddedDatabaseBuilder( backupDirectory ).setConfig( configParams ).newGraphDatabase();
```

Comment:
Why is a cast needed?

Reply:
Because the factory doesn't have that return type and a GraphDatabaseAPI is required to get access to the dependencies :man_shrugging: It happens like that in other areas of the code as well.

Reply:
It return a GraphDatabaseService. GraphDatabaseAPI is a subtype of GraphDatabaseService.

Probably can be improved with a strategically placed type parameter.

I'm making an addition to our refactoring column.